File: benchmarks/src/lua/word_count.lua
local M=500000;local seed=12345;local function ri(m)seed=(seed*16807)%2147483647;return seed%m end
local dict={};for i=1,2000 do dict[i]="w"..(i-1) end
local counts={};for i=1,2000 do counts[dict[i]]=0 end
local start=os.clock()
for i=1,M do local w=dict[ri(2000)+1];counts[w]=counts[w]+1 end
local mc=0;local mw=""
for i=1,2000 do local w=dict[i];local v=counts[w]
 if mc==0 then mc=v;mw=w elseif v>mc then mc=v;mw=w end end
local finish=os.clock()
print("Result: "..mc.."_"..mw);print(string.format("Time: %.0fms",(finish-start)*1000))

File: benchmarks/src/perl/word_count.pl
use Time::HiRes qw(time);
my$M=500000;my$seed=12345;my$ri=sub{my$m=shift;$seed=($seed*16807)%2147483647;$seed%$m};
my@dict;push@dict,"w$_"for 0..1999;my%counts;$counts{$_}=0 for@dict;
my$start=time();
$counts{$dict[$ri->(2000)]}++for 1..$M;
my($mc,$mw)=(0,"");for my$w(@dict){my$v=$counts{$w};if($mc==0){$mc=$v;$mw=$w}elsif($v>$mc){$mc=$v;$mw=$w}}
my$end=time();my$ms=($end-$start)*1000;print"Result: ${mc}_${mw}\n";printf"Time: %.0fms\n",$ms;
